Display info below map for markers appearing on screen

clinturbin opened the pull request:
For this branch:

- In index.html, removed placeholder div elements for listings
- In main.js, created function called showVisibleMarkers.  This function figures out which markers are currently shown on the map and then builds a div element to display the marker info in a list below the map.

Note: I needed to create 5 additional functions to build out the div.  Also, I still need to create a function to clear the current list before generating a new one.

@williammadisondavis @running-on-sunshine: Could you guys please review?

clinturbin committed:
Display info below map for markers appearing on screen.

williammadisondavis commented:
Looks very clean, with easy to identify variables. Good merge 👍🏼

running-on-sunshine commented on scripts/main.js:
```diff
@@ -31,18 +31,77 @@ var pins = [
+    var imageContainer = document.createElement("div");
+    imageContainer.classList.add("listing-image");
+    var listingImage = document.createElement("img");
+    listingImage.classList.add("listing-image");
```
Are both var imageContainer and var listingImage meant to have the same class "listing-image"?

clinturbin replied:
I gave them the same class name for now so that the image itself fills up the entire div container it is located in.  Looking at it, I think we might be able to get rid of the image container all together.

running-on-sunshine replied:
I think removing the image container would be fine, however have you tried seeing how the search listings would look like when expanded to different breakpoints (i.e. desktop screen size)? Would there be a possible need to have that image container later?

clinturbin committed:
Added function to clear current listings display

williammadisondavis reviewed (approved):
The updates look fine. Can we move ahead with this merge?

clinturbin commented:
Changes made:
- Added function clearListingDisplay, which clears the currently displayed elements before generating a new list of elements.
- Removed imageContainer div element from listingImageDisplay.  If we need to add this div element in later we can.

running-on-sunshine commented:
This is really good work! 🙂 You are good to go and may proceed to merge! 👍 

Thoughts: 

- Really liked how you broke the functions down into smaller pieces. 
- Again, you've nailed it with the variable names! Just descriptive of what they are and what they do.
- Re: listing image container. I agree with your assessment and decision! We'll proceed without the listing image container! If we find a need for it once we hit the desktop breakpoints, we'll just add it back in!
